backend/bufftracer/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import tracer, { Span } from 'dd-trace'
+import * as env from 'env-var'
+import { GraphQlRequest, RPCRequest } from './types'
+import { tagSpanTrace } from './tagSpanTrace'
+
+export * from './types'
+
+// Export dd-tracer for workers to do tracer.wrap
+export { tracer }
+
+const DD_ENABLE_TRACING = env.get('DD_ENABLE_TRACING').required().asBool()
+const DD_SERVICE_NAME = env
+  .get('DD_SERVICE_NAME')
+  .required(DD_ENABLE_TRACING)
+  .asString()
+
+const DD_TRACE_AGENT_HOSTNAME = env
+  .get('DD_TRACE_AGENT_HOSTNAME')
+  .required(DD_ENABLE_TRACING)
+  .asString()
+
+const DD_TRACE_AGENT_PORT = env
+  .get('DD_TRACE_AGENT_PORT')
+  .required(DD_ENABLE_TRACING)
+  .asString()
+
+const APP_STAGE = env.get('APP_STAGE').default('').asString()
+const NODE_ENV = env.get('NODE_ENV').asString()
+
+tracer.init({
+  enabled: DD_ENABLE_TRACING,
+  service: DD_SERVICE_NAME,
+  hostname: DD_TRACE_AGENT_HOSTNAME,
+  port: DD_TRACE_AGENT_PORT,
+  env: NODE_ENV || APP_STAGE,
+  logInjection: true,
+  trackAsyncScope: false, // As per https://github.com/DataDog/dd-trace-js/releases/tag/v0.16.0
+})
+
+tracer.use('express', {
+  hooks: {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    request: (
+      span: Span | undefined,
+      req: RPCRequest | GraphQlRequest,
+    ): void => {
+      tagSpanTrace(span, req)
+    },
+  },
+})
```

Comment:
@CoxyBoris, due to a bug within the dd-trace library, we can't export this as an `initTracer` function to use in our code. If we do, then the `tracer.use` hook won't work. It seems to be a known issue, so I will open a ticket for DD support and hope to change it back when I get back from vacation.

For now, we will simply do a `import from '@bufferapp/bufftracer' and won't be able to pass an error call back.

Reply:
That sounds good to me 🙂  , since the library catch errors and don't crash the process we're good